src/core/components/LanguageSwitcher/LanguageSwitcher.tsx

```diff
@@ -1,16 +1,54 @@
-import { ChangeEvent } from 'react';
+import { useState } from 'react';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import MenuProps from '@mui/material/Menu/Menu';
+import PublicIcon from '@mui/icons-material/Public';
+import { LANGUAGE } from '../../../config/i18n';
+import { useTranslation } from 'react-i18next';
+import './LanguageSwitcher.css';
 
 interface LanguageSwitcherProps {
   language: string;
-  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
+  onChange: (event: SelectChangeEvent<string>) => void;
 }
 
 export const LanguageSwitcher = ({ onChange, language }: LanguageSwitcherProps) => {
+  const { t } = useTranslation();
+  const [selectedLanguage, setSelectedLanguage] = useState(language);
+
+  const handleLanguageChange = (event: SelectChangeEvent<string>) => {
+    setSelectedLanguage(() => event.target.value);
+    onChange(event);
+  };
+
+  const menuProps: Partial<typeof MenuProps> = {
+    anchorOrigin: { vertical: 'bottom', horizontal: 'center' },
+    transformOrigin: { vertical: 'top', horizontal: 'center' },
+  };
+
   return (
-    <select value={language} onChange={onChange} id="language-switcher">
-      <option value="en">EN</option>
-      <option value="et">ET</option>
-      <option value="th">TH</option>
-    </select>
+    <FormControl className="language-switcher-container" variant="standard" sx={{ maxWidth: 100 }}>
+      <Select
+        MenuProps={menuProps}
+        className="language-switcher-select-standard"
+        value={selectedLanguage}
+        label={selectedLanguage}
+        onChange={handleLanguageChange}
+        disableUnderline={true}
+        IconComponent={PublicIcon}
+      >
+        {Object.keys(LANGUAGE).map((code: string) => (
```

Comment:
we would simplify this statement and the next ones, if we could export an array of lowercased languages from ../../../config/i18n

Reply:
Thanks for the hint! It works as you suggested now.